### PR TITLE
UCT/BUILD: Compile self transport regardless of knem.

### PR DESCRIPTION
**Pull request by yosefe**
This fixes "Device all is not available" message

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1115/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1116/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1118/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3048/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3052/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3054/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shssf pls take a look

File: src/ucp/core/ucp_context.c [hunk 1]
```diff
@@ -28,19 +28,26 @@ static const char *ucp_atomic_modes[] = {
     [UCP_ATOMIC_MODE_LAST]   = NULL,
 };
 
+static const char * ucp_device_type_names[] = {
+    [UCT_DEVICE_TYPE_NET]  = "network",
+    [UCT_DEVICE_TYPE_SHM]  = "intra-node",
+    [UCT_DEVICE_TYPE_ACC]  = "accelerator",
+    [UCT_DEVICE_TYPE_SELF] = "loopback",
+};
```
Comment (shamisp) [inline src/ucp/core/ucp_context.c]: So besides already existing name you are trying to add the dimension   of type ?  Does it mean that the uses has to specify the both ?

Comment (yosefe) [inline src/ucp/core/ucp_context.c]: the dimension already exists (in UCT), users can specify devices for each type.
this PR just add more verbose error message if devices are not available.

**Pull request merged**